src/wfs/src/main/java/org/geoserver/wfs/json/ComplexGeoJsonWriter.java

```diff
@@ -274,8 +293,8 @@ private void encodeLinkedFeatures(
     /** Encodes a list of features (chained features) as a JSON array. */
     private void encodeChainedFeatures(String attributeName, List<Feature> chainedFeatures) {
         // start the JSON object
-        jsonWriter.key(attributeName);
-        jsonWriter.array();
+        key(attributeName);
+        if (!isInsideArrayAttributeName(attributeName)) jsonWriter.array();
```

Comment:
I would have used an ``if`` with brackets and put a proper explaining what it means the branch being passed.

Reply:
Agreed, fixed